### DISCUSSION TITLE
Preventing `yarn add` from defaulting to `workspace:` in a monorepo?

### DISCUSSION THREAD

**Discussion by jdb8** [yarnpkg/berry#4498] [Q&A] [answered]
Hi there, I've taken a brief look at the docs, issues tab, and discussion list, but still may have missed where this has been addressed before - sorry if so.

We're using yarn 3.2.0 in a lerna monorepo and by default, when `yarn add`ing one monorepo package to another, yarn seems to default to `workspace:^`. Internally to the lockfile this is what we want: packages should be shared across the monorepo this way.

But currently our publishing script does not support the `workspace:` protocol at all (unlike `yarn version`), and so we're required to either manually adjust the generated `package.json` entry, use `yarn add -i` to pick a non-workspace version number, or avoid `yarn add` altogether.

Is there a configuration option or way we can have `yarn add` default to the actual version number present inside the monorepo package on disk? I looked at `enableTransparentWorkspaces: false`, but that seems like it'd remove the internal lockfile references, which isn't what we want - the desired outcome is to have things behave as if we'd explicitly run `yarn add internal-dep@^a.b.c`.

Thanks!

**Comment by arcanis** [comment, marked as answer]
No, it's not possible at the moment (unless you set the `-i` flag, iirc). Given it'd only be to accommodate third-party tooling that could easily add support for it, I'm not sure adding a configuration setting would be the right layer to fix the problem.

**Reply by jdb8** [reply]
Thanks for the quick response, that's what I assumed. I agree it's probably not worth adding this for the sake of legacy tooling.
